source/grammar/index.rst

```diff
@@ -80,53 +88,59 @@ repetition is denoted by curly brackets ``+`` (one or more) and ``*`` (zero or m
-    quantum_gate_signature: id quantum_gate_args id_lis
-    quantum_gate_args: ( "(" classical_type_and_id_list? ")" )?
+    quantum_gate_signature: id classical_params id_list
+    id_list: ( id "," )* id
+    index_type: id ( "[" expression "]" )?
+    index_list: ( index_type "," ) index_type
+    classical_params: ( "(" classical_type_and_id_list? ")" )?
     quantum_gate_block: "{" quantum_gate_call* "}"
     quantum_type_and_id_list: ( quantum_type_and_id "," )* quantum_type_and_id
-    quantum_type_and_id: quantum_type designator? association 
+    quantum_type_and_id: quantum_type designator? association
     quantum_statement: quantum_instruction ";"
     quantum_instruction: quantum_gate_call
         :| quantum_measurement
-    quantum_measurement: "measure" argument
+        :| barrier_instruction
+    quantum_measurement: "measure" quantum_declaration "->" bit_declaration
+        :| bit_declaration "= measure" quantum_declaration
+    barrier_instruction: "barrier" index_list
     quantum_gate_modifiers: ( "inv" | "pow" "[" nninteger "]" | "ctrl" ) "@"
-    quantum_gate_call: quantum_gate_name quantum_gate_args designator any_list ";"
-    quantum_gate_name: "CX" | "U" | "delay" | "reset" | id
+    quantum_gate_call: quantum_gate_name constant_args index_list | delay_call ";"
+    delay_call: "delay" ( "[" id | time_unit "]" )? ( range | index_list )
+    quantum_gate_name: "CX" | "U" | "reset" | id
         :| quantum_gate_modifier "@" quantum_gate_name
     quantum_gate_modifier: "inv" | "ctrl"
         :| "pow" nninteger?
-    quantum_gate_args: ( "(" expression_list? ")" )?
     quantum_declaration: quantum_type id designator
     quantum_type: "qubit" | "qreg"
     qubit_id_list: ( qubit_id "," )* qubit_id
-    qubit_id: id | "%" [ "q" ] nninteger
+    qubit_id: id | physical_qubit_id
+    physical_qubit_list: ( physical_qubit_id "," )* physical_qubit_id
+    physical_qubit_id: "%" [ "q" ] nninteger
     timing_box: "boxas" id gate_block
         :| "boxto" time_unit gate_block
     timing_type: "length" | "stretch" nninteger?
     time_terminator: time | "stretchinf"
-    time: id time_unit
+    time: id time_unit? | "lengthof(" id ")"
     time_unit: "dt" | "ns" | "us" | "ms" | "s"
-    calibration: calibration_grammar_declaration | calibration_definition
-    calibration_grammar_declaration: "defcalgrammar" id ";"
-    calibration_definition: "defcal" grammar_type id calibration_args qubit_id_list return_signature calibration_body
-    calibration_grammar_type: ( "openpulse" | id )*
-    calibration_args: ( "(" [ id_const_list ] ")" )*
-    mixed_type_specified_list: ( id_const_list | classical_type_and_id_list ) 
-        :| mixed_type_specified_list
-    id_const_list: ( expression_terminator "," )* expression_terminator
-    calibration_body: "{" ... "}"
\ No newline at end of file
+    calibration: cal_grammar_declaration | calibration_definition
+    cal_grammar_declaration: "defcalgrammar" cal_grammar ";"
```

Comment:
```suggestion
    cal_grammar_declaration: "defcalgrammar" cal_grammar_id ";"
```

Reply:
Addressed in https://github.com/taalexander/openqasm-1/pull/1/commits/782cb82ee9e49bb5067f7a409c0e03212d0acfc1.